test/mocks/authorize_net_mock.exs

```diff
@@ -54,6 +55,248 @@
+         "x-requested-with,cache-control,content-type,origin,method,SOAPAction"},
+        {"Access-Control-Allow-Credentials", "true"}, {"X-Cnection", "close"},
+        {"Content-Length", "577"}, {"Date", "Mon, 25 Dec 2017 17:08:12 GMT"},
+        {"Connection", "keep-alive"}],
+       request_url: "https://apitest.authorize.net/xml/v1/request.api",
+       status_code: 200}}
+    end
+    
+    def store_without_profile_fields do
+      {:ok,
+      %HTTPoison.Response{body: "﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><createCustomerProfileResponse xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns=\"AnetApi/xml/v1/schema/AnetApiSchema.xsd\"><messages><resultCode>Error</resultCode><message><code>E00041</code><text>One or more fields in the profile must contain a value.</text></message></messages></createCustomerProfileResponse>",
+       headers: [{"Cache-Control", "private"},
+        {"Content-Type", "application/xml; charset=utf-8"},
+        {"X-OPNET-Transaction-Trace",
+         "a2_b6b84b43-d399-4dde-bc12-fb1f8ccf4b27-51156-15831457"},
+        {"Access-Control-Allow-Origin", "*"},
+        {"Access-Control-Allow-Methods", "PUT,OPTIONS,POST,GET"},
+        {"Access-Control-Allow-Headers",
+         "x-requested-with,cache-control,content-type,origin,method,SOAPAction"},
+        {"Access-Control-Allow-Credentials", "true"}, {"X-Cnection", "close"},
+        {"Content-Length", "408"}, {"Date", "Mon, 25 Dec 2017 17:12:30 GMT"},
+        {"Connection", "keep-alive"}],
+       request_url: "https://apitest.authorize.net/xml/v1/request.api",
+       status_code: 200}} 
+    end
+
+    #unstore mock response
+    def successful_unstore_response do
+      {:ok,
+      %HTTPoison.Response{body: "﻿<?xml version=\"1.0\" encoding=\"utf-8\"?><deleteCustomerProfileResponse xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns=\"AnetApi/xml/v1/schema/AnetApiSchema.xsd\"><messages><resultCode>Ok</resultCode><message><code>I00001</code><text>Successful.</text></message></messages></deleteCustomerProfileResponse>",
+       headers: [{"Cache-Control", "private"},
+        {"Content-Type", "application/xml; charset=utf-8"},
+        {"X-OPNET-Transaction-Trace",
+         "a2_b6b84b43-d399-4dde-bc12-fb1f8ccf4b27-51156-15833786"},
+        {"Access-Control-Allow-Origin", "*"},
+        {"Access-Control-Allow-Methods", "PUT,OPTIONS,POST,GET"},
+        {"Access-Control-Allow-Headers",
+         "x-requested-with,cache-control,content-type,origin,method,SOAPAction"},
+        {"Access-Control-Allow-Credentials", "true"}, {"X-Cnection", "close"},
+        {"Content-Length", "361"}, {"Date", "Mon, 25 Dec 2017 17:21:20 GMT"},
+        {"Connection", "keep-alive"}],
+       request_url: "https://apitest.authorize.net/xml/v1/request.api",
+       status_code: 200}}     
+    end
+
     def network_error_response do
```

Comment:
I'm just copying my previous comment (even though you've removed the test for now).
Your `respond` method only partially matches a network timeout, `HTTPoison.request` returns:
```elixir
{:error, %HTTPoison.Error{__exception__: term, id: reference | nil, reason: any}}
```
for all errors including timeouts. Also see #30.

So your `body` and `status_code` will be `nil`. Essentially this function does not really mock the scenario.